<a href="https://colab.research.google.com/github/mxrxdth/mxrxdth001/blob/master/Old_Toronto_v_s_Manhattan_v_s_Paris.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## Table of Contents


1. Introduction/Business Problem

2. Data

3. Report
   


# 1. Introduction/Business Problem



**(1). The problem & discussion of the background.**

Is New York City more like Toronto or Paris?
If a NYC-based brand would like to expand its business to overseas especially the brick-and-mortar, whether they should expect similarity or dissimilarity in its options of cities?
 
It would be interesting to compare the neighborhoods of these multicultural cities as financial capital of different countries and determine how similar or dissimilar they are. However, While the area of Paris is 40.7 mi² with its 20 ‎Arrondissements, in our data New York City is 302.6 mi² with 302 neighborhoods and Toronto 243.3 mi² with 98 neighborhoods.
Old Toronto (https://en.wikipedia.org/wiki/Toronto) and Manhattan (22.82 mi²) as the most densely populated part of the respective city, should be more comparable.

Therefore, our problem is refined as: Is Manhattan more like core Toronto or Paris?

<br/>

**(2). A description of the data & how it will be used to solve the problem.**

Source of city neighborhoods, geo coordinates as follows.

Toronto: http://cocl.us/Geospatial_data, https://en.wikipedia.org/wiki/List_of_postal_codes_of_Canada:_M

New York City: https://cocl.us/new_york_dataset

Paris: http://download.geonames.org/export/zip/FR.zip, processed in https://docs.google.com/spreadsheets/d/e/2PACX-1vQXNcKhCQi7jy0AoRKKAafKRqZsLJhwC8jfgBrdHLOqFiPo09A8IHb6C-GpNWI21g/pub?output=xlsx
 
Based on the geo coordinates of neighborhoods in the three cities, we get the data of venues around each neighborhood via Foursquare API.
The Core Toronto data is filtered out from Toronto data, Manhattan out of New York City, Paris as it is.

<br/>
<br/>

# 2. Data

In [1]:
import numpy as np # library to handle data in a vectorized manner
import pandas as pd # library for data analsysis
pd.set_option('display.max_columns', None)
pd.set_option('display.max_rows', None)

import json # library to handle JSON files

!conda install -c conda-forge --no-deps altair --yes
!conda install -c conda-forge --no-deps vincent --yes

#!conda install -c conda-forge geopy --yes # uncomment this line if you haven't completed the Foursquare API lab
!conda install -c conda-forge geopy=1.19.0 --yes
from geopy.geocoders import Nominatim # convert an address into latitude and longitude values

import requests # library to handle requests
from pandas.io.json import json_normalize # tranform JSON file into a pandas dataframe

# Matplotlib and associated plotting modules
import matplotlib.cm as cm
import matplotlib.colors as colors

# import k-means from clustering stage
from sklearn.cluster import KMeans

#!conda install -c conda-forge folium=0.5.0 --yes # uncomment this line if you haven't completed the Foursquare API lab
!conda install -c conda-forge --no-deps folium=0.5.0 --yes
import folium # map rendering library

print('Libraries imported.')

/bin/bash: conda: command not found
/bin/bash: conda: command not found
/bin/bash: conda: command not found
/bin/bash: conda: command not found
Libraries imported.


In [2]:
!pip install branca
import folium
print('folium imported!')

folium imported!


## 1) Old Toronto

In [5]:
 # The webpage URL whose table we want to extract 
url_toronto_codes = "https://en.wikipedia.org/wiki/List_of_postal_codes_of_Canada:_M"
  
# Assign the table data to a Pandas dataframe 
df_toronto = pd.read_html(url_toronto_codes)
df_toronto = df_toronto[0]
df_toronto.head()

Postal code           Borough                Neighborhood
0         M1A      Not assigned                         NaN
1         M2A      Not assigned                         NaN
2         M3A        North York                   Parkwoods
3         M4A        North York            Victoria Village
4         M5A  Downtown Toronto  Regent Park / Harbourfront

In [6]:
df_toronto.isnull().sum()

Postal code      0
Borough          0
Neighborhood    77
dtype: int64

In [7]:
df_toronto = df_toronto[df_toronto['Borough'] != 'Not assigned'].reset_index(drop=True)
df_toronto.rename(columns={"Postal code": "Postal Code"}, inplace=True)
df_toronto.head()

Postal Code           Borough                                  Neighborhood
0         M3A        North York                                     Parkwoods
1         M4A        North York                              Victoria Village
2         M5A  Downtown Toronto                    Regent Park / Harbourfront
3         M6A        North York             Lawrence Manor / Lawrence Heights
4         M7A  Downtown Toronto  Queen's Park / Ontario Provincial Government

In [8]:
df_toronto.isnull().sum()

Postal Code     0
Borough         0
Neighborhood    0
dtype: int64

In [9]:
df_toronto_grouped = df_toronto.groupby('Postal Code').agg({'Borough':'first', 'Neighborhood':'sum'}).reset_index()
df_toronto_grouped = df_toronto_grouped.apply(lambda x: x.str.replace(' /',','))
df_toronto_grouped.head()

Postal Code      Borough                            Neighborhood
0         M1B  Scarborough                          Malvern, Rouge
1         M1C  Scarborough  Rouge Hill, Port Union, Highland Creek
2         M1E  Scarborough       Guildwood, Morningside, West Hill
3         M1G  Scarborough                                  Woburn
4         M1H  Scarborough                               Cedarbrae

In [10]:
df_toronto_grouped.shape

(103, 3)

In [11]:
url_toronto = "http://cocl.us/Geospatial_data"
df_toronto_coordinates = pd.read_csv(url_toronto)
df_toronto_coordinates

Postal Code   Latitude  Longitude
0           M1B  43.806686 -79.194353
1           M1C  43.784535 -79.160497
2           M1E  43.763573 -79.188711
3           M1G  43.770992 -79.216917
4           M1H  43.773136 -79.239476
5           M1J  43.744734 -79.239476
6           M1K  43.727929 -79.262029
7           M1L  43.711112 -79.284577
8           M1M  43.716316 -79.239476
9           M1N  43.692657 -79.264848
10          M1P  43.757410 -79.273304
11          M1R  43.750072 -79.295849
12          M1S  43.794200 -79.262029
13          M1T  43.781638 -79.304302
14          M1V  43.815252 -79.284577
15          M1W  43.799525 -79.318389
16          M1X  43.836125 -79.205636
17          M2H  43.803762 -79.363452
18          M2J  43.778517 -79.346556
19          M2K  43.786947 -79.385975
20          M2L  43.757490 -79.374714
21          M2M  43.789053 -79.408493
22          M2N  43.770120 -79.408493
23          M2P  43.752758 -79.400049
24          M2R  43.782736 -79.442259
25          M3A  43.753259 -79.329656
26          M3B  43.745906 -79.352188
27          M3C  43.725900 -79.340923
28          M3H  43.754328 -79.442259
29          M3J  43.767980 -79.487262
30          M3K  43.737473 -79.464763
31          M3L  43.739015 -79.506944
32          M3M  43.728496 -79.495697
33          M3N  43.761631 -79.520999
34          M4A  43.725882 -79.315572
35          M4B  43.706397 -79.309937
36          M4C  43.695344 -79.318389
37          M4E  43.676357 -79.293031
38          M4G  43.709060 -79.363452
39          M4H  43.705369 -79.349372
40          M4J  43.685347 -79.338106
41          M4K  43.679557 -79.352188
42          M4L  43.668999 -79.315572
43          M4M  43.659526 -79.340923
44          M4N  43.728020 -79.388790
45          M4P  43.712751 -79.390197
46          M4R  43.715383 -79.405678
47          M4S  43.704324 -79.388790
48          M4T  43.689574 -79.383160
49          M4V  43.686412 -79.400049
50          M4W  43.679563 -79.377529
51          M4X  43.667967 -79.367675
52          M4Y  43.665860 -79.383160
53          M5A  43.654260 -79.360636
54          M5B  43.657162 -79.378937
55          M5C  43.651494 -79.375418
56          M5E  43.644771 -79.373306
57          M5G  43.657952 -79.387383
58          M5H  43.650571 -79.384568
59          M5J  43.640816 -79.381752
60          M5K  43.647177 -79.381576
61          M5L  43.648198 -79.379817
62          M5M  43.733283 -79.419750
63          M5N  43.711695 -79.416936
64          M5P  43.696948 -79.411307
65          M5R  43.672710 -79.405678
66          M5S  43.662696 -79.400049
67          M5T  43.653206 -79.400049
68          M5V  43.628947 -79.394420
69          M5W  43.646435 -79.374846
70          M5X  43.648429 -79.382280
71          M6A  43.718518 -79.464763
72          M6B  43.709577 -79.445073
73          M6C  43.693781 -79.428191
74          M6E  43.689026 -79.453512
75          M6G  43.669542 -79.422564
76          M6H  43.669005 -79.442259
77          M6J  43.647927 -79.419750
78          M6K  43.636847 -79.428191
79          M6L  43.713756 -79.490074
80          M6M  43.691116 -79.476013
81          M6N  43.673185 -79.487262
82          M6P  43.661608 -79.464763
83          M6R  43.648960 -79.456325
84          M6S  43.651571 -79.484450
85          M7A  43.662301 -79.389494
86          M7R  43.636966 -79.615819
87          M7Y  43.662744 -79.321558
88          M8V  43.605647 -79.501321
89          M8W  43.602414 -79.543484
90          M8X  43.653654 -79.506944
91          M8Y  43.636258 -79.498509
92          M8Z  43.628841 -79.520999
93          M9A  43.667856 -79.532242
94          M9B  43.650943 -79.554724
95          M9C  43.643515 -79.577201
96          M9L  43.756303 -79.565963
97          M9M  43.724766 -79.532242
98          M9N  43.706876 -79.518188
99          M9P  43.696319 -79.532242
100         M9R  43.688905 -79.554724
101         M9V  43.739416 -79.588437
102         M9W  43.706748 -79.594054

In [12]:
df_toronto_grouped.columns[0]==df_toronto_coordinates.columns[0]

True

In [13]:
toronto_data = df_toronto_grouped.join(df_toronto_coordinates.set_index('Postal Code'), on='Postal Code')
toronto_data.head(10)

Postal Code      Borough                                     Neighborhood  \
0         M1B  Scarborough                                   Malvern, Rouge   
1         M1C  Scarborough           Rouge Hill, Port Union, Highland Creek   
2         M1E  Scarborough                Guildwood, Morningside, West Hill   
3         M1G  Scarborough                                           Woburn   
4         M1H  Scarborough                                        Cedarbrae   
5         M1J  Scarborough                              Scarborough Village   
6         M1K  Scarborough      Kennedy Park, Ionview, East Birchmount Park   
7         M1L  Scarborough                  Golden Mile, Clairlea, Oakridge   
8         M1M  Scarborough  Cliffside, Cliffcrest, Scarborough Village West   
9         M1N  Scarborough                      Birch Cliff, Cliffside West   

    Latitude  Longitude  
0  43.806686 -79.194353  
1  43.784535 -79.160497  
2  43.763573 -79.188711  
3  43.770992 -79.216917  
4  43.773136 -79.239476  
5  43.744734 -79.239476  
6  43.727929 -79.262029  
7  43.711112 -79.284577  
8  43.716316 -79.239476  
9  43.692657 -79.264848

In [14]:
print('There are {} neighborhoods in Toronto.'.format(len(toronto_data['Neighborhood'].unique())))

There are 98 neighborhoods in Toronto.


In [15]:
#
old_toronto_data = toronto_data[toronto_data['Borough'].str.contains('Toronto')].reset_index(drop=True)
old_toronto_data

Postal Code           Borough  \
0          M4E      East Toronto   
1          M4K      East Toronto   
2          M4L      East Toronto   
3          M4M      East Toronto   
4          M4N   Central Toronto   
5          M4P   Central Toronto   
6          M4R   Central Toronto   
7          M4S   Central Toronto   
8          M4T   Central Toronto   
9          M4V   Central Toronto   
10         M4W  Downtown Toronto   
11         M4X  Downtown Toronto   
12         M4Y  Downtown Toronto   
13         M5A  Downtown Toronto   
14         M5B  Downtown Toronto   
15         M5C  Downtown Toronto   
16         M5E  Downtown Toronto   
17         M5G  Downtown Toronto   
18         M5H  Downtown Toronto   
19         M5J  Downtown Toronto   
20         M5K  Downtown Toronto   
21         M5L  Downtown Toronto   
22         M5N   Central Toronto   
23         M5P   Central Toronto   
24         M5R   Central Toronto   
25         M5S  Downtown Toronto   
26         M5T  Downtown Toronto   
27         M5V  Downtown Toronto   
28         M5W  Downtown Toronto   
29         M5X  Downtown Toronto   
30         M6G  Downtown Toronto   
31         M6H      West Toronto   
32         M6J      West Toronto   
33         M6K      West Toronto   
34         M6P      West Toronto   
35         M6R      West Toronto   
36         M6S      West Toronto   
37         M7A  Downtown Toronto   
38         M7Y      East Toronto   

                                         Neighborhood   Latitude  Longitude  
0                                         The Beaches  43.676357 -79.293031  
1                        The Danforth West, Riverdale  43.679557 -79.352188  
2                      India Bazaar, The Beaches West  43.668999 -79.315572  
3                                     Studio District  43.659526 -79.340923  
4                                       Lawrence Park  43.728020 -79.388790  
5                                    Davisville North  43.712751 -79.390197  
6                                  North Toronto West  43.715383 -79.405678  
7                                          Davisville  43.704324 -79.388790  
8                         Moore Park, Summerhill East  43.689574 -79.383160  
9   Summerhill West, Rathnelly, South Hill, Forest...  43.686412 -79.400049  
10                                           Rosedale  43.679563 -79.377529  
11                        St. James Town, Cabbagetown  43.667967 -79.367675  
12                               Church and Wellesley  43.665860 -79.383160  
13                          Regent Park, Harbourfront  43.654260 -79.360636  
14                           Garden District, Ryerson  43.657162 -79.378937  
15                                     St. James Town  43.651494 -79.375418  
16                                        Berczy Park  43.644771 -79.373306  
17                                 Central Bay Street  43.657952 -79.387383  
18                           Richmond, Adelaide, King  43.650571 -79.384568  
19  Harbourfront East, Union Station, Toronto Islands  43.640816 -79.381752  
20           Toronto Dominion Centre, Design Exchange  43.647177 -79.381576  
21                     Commerce Court, Victoria Hotel  43.648198 -79.379817  
22                                           Roselawn  43.711695 -79.416936  
23                           Forest Hill North & West  43.696948 -79.411307  
24                The Annex, North Midtown, Yorkville  43.672710 -79.405678  
25                     University of Toronto, Harbord  43.662696 -79.400049  
26          Kensington Market, Chinatown, Grange Park  43.653206 -79.400049  
27  CN Tower, King and Spadina, Railway Lands, Har...  43.628947 -79.394420  
28                                     Stn A PO Boxes  43.646435 -79.374846  
29             First Canadian Place, Underground city  43.648429 -79.382280  
30                                           Christie  43.669542 -79.422564  
31                       Dufferin, Dovercourt Village  43.669005 -

### Foursquare

In [16]:
CLIENT_ID = 'CLDHWKFPZ03KEUOBPFAL2O3YCRHPJ0MOSJLPJXFBIY4EHVAI' # your Foursquare ID
CLIENT_SECRET = '3EYLGLVIXUOV2WIIA5M3CAROIPGFZZKV1QO3QUQQSUHLNNIT' # your Foursquare Secret
VERSION = '20180605' # Foursquare API version
LIMIT = 100

print('Your credentails:')
print('CLIENT_ID: ' + CLIENT_ID)
print('CLIENT_SECRET:' + CLIENT_SECRET)

Your credentails:
CLIENT_ID: CLDHWKFPZ03KEUOBPFAL2O3YCRHPJ0MOSJLPJXFBIY4EHVAI
CLIENT_SECRET:3EYLGLVIXUOV2WIIA5M3CAROIPGFZZKV1QO3QUQQSUHLNNIT


### Create a function to repeat the same process to all the neighborhoods in the cities

In [0]:
def getNearbyVenues(names, latitudes, longitudes, radius):
    
    venues_list=[]
    for name, lat, lng in zip(names, latitudes, longitudes):
        print(name)
            
        # create the API request URL
        url = 'https://api.foursquare.com/v2/venues/explore?&client_id={}&client_secret={}&v={}&ll={},{}&radius={}&limit={}'.format(
            CLIENT_ID, 
            CLIENT_SECRET, 
            VERSION, 
            lat, 
            lng, 
            radius, 
            LIMIT)
            
        # make the GET request
        results = requests.get(url).json()["response"]['groups'][0]['items']
        
        # return only relevant information for each nearby venue
        venues_list.append([(
            name, 
            lat, 
            lng, 
            v['venue']['name'], 
            v['venue']['location']['lat'], 
            v['venue']['location']['lng'],  
            v['venue']['categories'][0]['name']) for v in results])

    nearby_venues = pd.DataFrame([item for venue_list in venues_list for item in venue_list])
    nearby_venues.columns = ['Neighborhood', 
                  'Neighborhood Latitude', 
                  'Neighborhood Longitude', 
                  'Venue', 
                  'Venue Latitude', 
                  'Venue Longitude', 
                  'Venue Category']
    
    return(nearby_venues)

In [18]:
old_toronto_venues = getNearbyVenues(names=old_toronto_data['Neighborhood'],
                                   latitudes=old_toronto_data['Latitude'],
                                   longitudes=old_toronto_data['Longitude'],
                                    radius=600
                                  )

The Beaches
The Danforth West, Riverdale
India Bazaar, The Beaches West
Studio District
Lawrence Park
Davisville North
North Toronto West
Davisville
Moore Park, Summerhill East
Summerhill West, Rathnelly, South Hill, Forest Hill SE, Deer Park
Rosedale
St. James Town, Cabbagetown
Church and Wellesley
Regent Park, Harbourfront
Garden District, Ryerson
St. James Town
Berczy Park
Central Bay Street
Richmond, Adelaide, King
Harbourfront East, Union Station, Toronto Islands
Toronto Dominion Centre, Design Exchange
Commerce Court, Victoria Hotel
Roselawn
Forest Hill North & West
The Annex, North Midtown, Yorkville
University of Toronto, Harbord
Kensington Market, Chinatown, Grange Park
CN Tower, King and Spadina, Railway Lands, Harbourfront West, Bathurst  Quay, South Niagara, Island airport
Stn A PO Boxes
First Canadian Place, Underground city
Christie
Dufferin, Dovercourt Village
Little Portugal, Trinity
Brockton, Parkdale Village, Exhibition Place
High Park, The Junction South
Parkdale, Ro

In [19]:
print(old_toronto_venues.shape)
old_toronto_venues.head()

(2122, 7)


Neighborhood  Neighborhood Latitude  Neighborhood Longitude  \
0  The Beaches              43.676357              -79.293031   
1  The Beaches              43.676357              -79.293031   
2  The Beaches              43.676357              -79.293031   
3  The Beaches              43.676357              -79.293031   
4  The Beaches              43.676357              -79.293031   

               Venue  Venue Latitude  Venue Longitude  \
0  Glen Manor Ravine       43.676821       -79.293942   
1    Tori's Bakeshop       43.672114       -79.290331   
2  Beaches Bake Shop       43.680363       -79.289692   
3     The Beech Tree       43.680493       -79.288846   
4    Ed's Real Scoop       43.672630       -79.287993   

                  Venue Category  
0                          Trail  
1  Vegetarian / Vegan Restaurant  
2                         Bakery  
3                      Gastropub  
4                 Ice Cream Shop

In [20]:
old_toronto_venues.groupby('Neighborhood').count()

Neighborhood Latitude  \
Neighborhood                                                                
Berczy Park                                                            90   
Brockton, Parkdale Village, Exhibition Place                           40   
Business reply mail Processing CentrE                                  25   
CN Tower, King and Spadina, Railway Lands, Harb...                     18   
Central Bay Street                                                    100   
Christie                                                               20   
Church and Wellesley                                                  100   
Commerce Court, Victoria Hotel                                        100   
Davisville                                                             47   
Davisville North                                                       13   
Dufferin, Dovercourt Village                                           20   
First Canadian Place, Underground city                                100   
Forest Hill North & West                                                4   
Garden District, Ryerson                                              100   
Harbourfront East, Union Station, Toronto Islands                     100   
High Park, The Junction South                                          39   
India Bazaar, The Beaches West                                         33   
Kensington Market, Chinatown, Grange Park                             100   
Lawrence Park                                                           4   
Little Portugal, Trinity                                               65   
Moore Park, Summerhill East                                             4   
North Toronto West                                                     25   
Parkdale, Roncesvalles                                                 31   
Queen's Park, Ontario Provincial Government                            57   
Regent Park, Harbourfront                                              58   
Richmond, Adelaide, King                                              100   
Rosedale                                                                5   
Roselawn                                                                5   
Runnymede, Swansea                                                     53   
St. James Town                                                        100   
St. James Town, Cabbagetown                                            56   
Stn A PO Boxes                                                        100   
Studio District                                                        68   
Summerhill West, Rathnelly, South Hill, Forest ...                     50   
The Annex, North Midtown, Yorkville                                    32   
The Beaches                                                            20   
The Danforth West, Riverdale                                           66   
Toronto Dominion Centre, Design Exchange                              100   
University of Toronto, Harbord                                         74   

                                                    Neighborhood Longitude  \
Neighborhood                                                                 
Berczy Park                                                             90   
Brockton, Parkdale Village, Exhibition Place                            40   
Business reply mail Processing CentrE                                   25   
CN Tower, King and Spadina, Railway Lands, Harb...                      18   
Central Bay Street                                                     100   
Christie                                                                20   
Church and Wellesley                                                   100   
Commerce Court, Victoria Hotel                                         100   
Davisville                                                              47   
Davisville North                    

In [21]:
print('There are {} uniques categories.'.format(len(old_toronto_venues['Venue Category'].unique())))

There are 249 uniques categories.


In [22]:
print('There are {} neighborhoods in Old Toronto.'.format(len(old_toronto_venues['Neighborhood'].unique())))

There are 39 neighborhoods in Old Toronto.


In [23]:
# one hot encoding
old_toronto_onehot = pd.get_dummies(old_toronto_venues['Venue Category'], ).drop('Neighborhood', axis=1)
old_toronto_onehot.head()

Afghan Restaurant  Airport  Airport Food Court  Airport Gate  \
0                  0        0                   0             0   
1                  0        0                   0             0   
2                  0        0                   0             0   
3                  0        0                   0             0   
4                  0        0                   0             0   

   Airport Lounge  Airport Service  Airport Terminal  American Restaurant  \
0               0                0                 0                    0   
1               0                0                 0                    0   
2               0                0                 0                    0   
3               0                0                 0                    0   
4               0                0                 0                    0   

   Amphitheater  Animal Shelter  Antique Shop  Aquarium  Arepa Restaurant  \
0             0               0             0         0                 0   
1             0               0             0         0                 0   
2             0               0             0         0                 0   
3             0               0             0         0                 0   
4             0               0             0         0                 0   

   Art Gallery  Arts & Crafts Store  Asian Restaurant  Athletics & Sports  \
0            0                    0                 0                   0   
1            0                    0                 0                   0   
2            0                    0                 0                   0   
3            0                    0                 0                   0   
4            0                    0                 0                   0   

   BBQ Joint  Baby Store  Bagel Shop  Bakery  Bank  Bar  Baseball Stadium  \
0          0           0           0       0     0    0                 0   
1          0           0           0       0     0    0                 0   
2          0           0           0       1     0    0                 0   
3          0           0           0       0     0    0                 0   
4          0           0           0       0     0    0                 0   

   Basketball Stadium  Beach  Beer Bar  Beer Store  Belgian Restaurant  \
0                   0      0         0           0                   0   
1                   0      0         0           0                   0   
2                   0      0         0           0                   0   
3                   0      0         0           0                   0   
4                   0      0         0           0                   0   

   Bistro  Boat or Ferry  Bookstore  Boutique  Brazilian Restaurant  \
0       0              0          0         0                     0   
1       0              0          0         0                     0   
2       0              0          0         0                     0   
3       0              0          0         0                     0   
4       0              0          0         0                     0   

   Breakfast Spot  Brewery  Bubble Tea Shop  Burger Joint  Burrito Place  \
0               0        0                0             0              0   
1               0        0                0             0              0   
2               0        0                0             0              0   
3               0        0                0             0              0   
4               0        0                0             0              0   

   Bus Line  Butcher  Café  Cajun / Creole Restaurant  Camera Store  \
0         0        0     0                          0             0   
1         0        0     0                          0             0   
2         0        0     0                          0             0   
3         0        0     0                          0             0   
4         0        0     0                          0           

In [24]:
# add neighborhood column back to dataframe
old_toronto_onehot.insert(0, 'Neighborhood', old_toronto_venues['Neighborhood'], True)
old_toronto_onehot.head()

Neighborhood  Afghan Restaurant  Airport  Airport Food Court  Airport Gate  \
0  The Beaches                  0        0                   0             0   
1  The Beaches                  0        0                   0             0   
2  The Beaches                  0        0                   0             0   
3  The Beaches                  0        0                   0             0   
4  The Beaches                  0        0                   0             0   

   Airport Lounge  Airport Service  Airport Terminal  American Restaurant  \
0               0                0                 0                    0   
1               0                0                 0                    0   
2               0                0                 0                    0   
3               0                0                 0                    0   
4               0                0                 0                    0   

   Amphitheater  Animal Shelter  Antique Shop  Aquarium  Arepa Restaurant  \
0             0               0             0         0                 0   
1             0               0             0         0                 0   
2             0               0             0         0                 0   
3             0               0             0         0                 0   
4             0               0             0         0                 0   

   Art Gallery  Arts & Crafts Store  Asian Restaurant  Athletics & Sports  \
0            0                    0                 0                   0   
1            0                    0                 0                   0   
2            0                    0                 0                   0   
3            0                    0                 0                   0   
4            0                    0                 0                   0   

   BBQ Joint  Baby Store  Bagel Shop  Bakery  Bank  Bar  Baseball Stadium  \
0          0           0           0       0     0    0                 0   
1          0           0           0       0     0    0                 0   
2          0           0           0       1     0    0                 0   
3          0           0           0       0     0    0                 0   
4          0           0           0       0     0    0                 0   

   Basketball Stadium  Beach  Beer Bar  Beer Store  Belgian Restaurant  \
0                   0      0         0           0                   0   
1                   0      0         0           0                   0   
2                   0      0         0           0                   0   
3                   0      0         0           0                   0   
4                   0      0         0           0                   0   

   Bistro  Boat or Ferry  Bookstore  Boutique  Brazilian Restaurant  \
0       0              0          0         0                     0   
1       0              0          0         0                     0   
2       0              0          0         0                     0   
3       0              0          0         0                     0   
4       0              0          0         0                     0   

   Breakfast Spot  Brewery  Bubble Tea Shop  Burger Joint  Burrito Place  \
0               0        0                0             0              0   
1               0        0                0             0              0   
2               0        0                0             0              0   
3               0        0                0             0              0   
4               0        0                0             0              0   

   Bus Line  Butcher  Café  Cajun / Creole Restaurant  Camera Store  \
0         0        0     0                          0             0   
1         0        0     0                          0             0   
2         0        0     0                          0             0   
3         0        0     0                          0   

In [25]:
'Neiborhood'in (old_toronto_venues['Venue Category'].unique())

False

## 2) Manhattan

In [26]:
!wget -q -O 'newyork_data.json' https://cocl.us/new_york_dataset
print('Data downloaded!')

Data downloaded!


In [0]:
with open('newyork_data.json') as json_data:
    newyork_data = json.load(json_data)

In [0]:
neighborhoods_data = newyork_data['features']

In [29]:
neighborhoods_data[0]

{'geometry': {'coordinates': [-73.84720052054902, 40.89470517661],
  'type': 'Point'},
 'geometry_name': 'geom',
 'id': 'nyu_2451_34572.1',
 'properties': {'annoangle': 0.0,
  'annoline1': 'Wakefield',
  'annoline2': None,
  'annoline3': None,
  'bbox': [-73.84720052054902,
   40.89470517661,
   -73.84720052054902,
   40.89470517661],
  'borough': 'Bronx',
  'name': 'Wakefield',
  'stacked': 1},
 'type': 'Feature'}

In [0]:
# define the dataframe columns
column_names = ['Borough', 'Neighborhood', 'Latitude', 'Longitude'] 

# instantiate the dataframe
neighborhoods = pd.DataFrame(columns=column_names)

In [31]:
neighborhoods

Empty DataFrame
Columns: [Borough, Neighborhood, Latitude, Longitude]
Index: []

In [0]:
for data in neighborhoods_data:
    borough = neighborhood_name = data['properties']['borough'] 
    neighborhood_name = data['properties']['name']
        
    neighborhood_latlon = data['geometry']['coordinates']
    neighborhood_lat = neighborhood_latlon[1]
    neighborhood_lon = neighborhood_latlon[0]
    
    neighborhoods = neighborhoods.append({'Borough': borough,
                                          'Neighborhood': neighborhood_name,
                                          'Latitude': neighborhood_lat,
                                          'Longitude': neighborhood_lon}, ignore_index=True)

In [33]:
neighborhoods.head()

Borough Neighborhood   Latitude  Longitude
0   Bronx    Wakefield  40.894705 -73.847201
1   Bronx   Co-op City  40.874294 -73.829939
2   Bronx  Eastchester  40.887556 -73.827806
3   Bronx    Fieldston  40.895437 -73.905643
4   Bronx    Riverdale  40.890834 -73.912585

In [34]:
print('There are {} neighborhoods in New York City.'.format(len(neighborhoods['Neighborhood'].unique())))

There are 302 neighborhoods in New York City.


In [35]:
manhattan_data = neighborhoods[neighborhoods['Borough'] == 'Manhattan'].reset_index(drop=True)
manhattan_data.head()

Borough        Neighborhood   Latitude  Longitude
0  Manhattan         Marble Hill  40.876551 -73.910660
1  Manhattan           Chinatown  40.715618 -73.994279
2  Manhattan  Washington Heights  40.851903 -73.936900
3  Manhattan              Inwood  40.867684 -73.921210
4  Manhattan    Hamilton Heights  40.823604 -73.949688

In [36]:
manhattan_venues = getNearbyVenues(names=manhattan_data['Neighborhood'],
                                   latitudes=manhattan_data['Latitude'],
                                   longitudes=manhattan_data['Longitude'],
                                   radius=500
                                  )

Marble Hill
Chinatown
Washington Heights
Inwood
Hamilton Heights
Manhattanville
Central Harlem
East Harlem
Upper East Side
Yorkville
Lenox Hill
Roosevelt Island
Upper West Side
Lincoln Square
Clinton
Midtown
Murray Hill
Chelsea
Greenwich Village
East Village
Lower East Side
Tribeca
Little Italy
Soho
West Village
Manhattan Valley
Morningside Heights
Gramercy
Battery Park City
Financial District
Carnegie Hill
Noho
Civic Center
Midtown South
Sutton Place
Turtle Bay
Tudor City
Stuyvesant Town
Flatiron
Hudson Yards


In [37]:
print(manhattan_venues.shape)
manhattan_venues.head()

(3279, 7)


Neighborhood  Neighborhood Latitude  Neighborhood Longitude          Venue  \
0  Marble Hill              40.876551               -73.91066       Arturo's   
1  Marble Hill              40.876551               -73.91066    Bikram Yoga   
2  Marble Hill              40.876551               -73.91066  Tibbett Diner   
3  Marble Hill              40.876551               -73.91066        Dunkin'   
4  Marble Hill              40.876551               -73.91066      Starbucks   

   Venue Latitude  Venue Longitude Venue Category  
0       40.874412       -73.910271    Pizza Place  
1       40.876844       -73.906204    Yoga Studio  
2       40.880404       -73.908937          Diner  
3       40.877136       -73.906666     Donut Shop  
4       40.877531       -73.905582    Coffee Shop

In [38]:
manhattan_venues.groupby('Neighborhood').count()

Neighborhood Latitude  Neighborhood Longitude  Venue  \
Neighborhood                                                                
Battery Park City                       87                      87     87   
Carnegie Hill                           93                      93     93   
Central Harlem                          42                      42     42   
Chelsea                                100                     100    100   
Chinatown                              100                     100    100   
Civic Center                           100                     100    100   
Clinton                                100                     100    100   
East Harlem                             39                      39     39   
East Village                           100                     100    100   
Financial District                     100                     100    100   
Flatiron                               100                     100    100   
Gramercy                                96                      96     96   
Greenwich Village                      100                     100    100   
Hamilton Heights                        61                      61     61   
Hudson Yards                            78                      78     78   
Inwood                                  56                      56     56   
Lenox Hill                             100                     100    100   
Lincoln Square                         100                     100    100   
Little Italy                           100                     100    100   
Lower East Side                         57                      57     57   
Manhattan Valley                        47                      47     47   
Manhattanville                          44                      44     44   
Marble Hill                             25                      25     25   
Midtown                                100                     100    100   
Midtown South                          100                     100    100   
Morningside Heights                     40                      40     40   
Murray Hill                            100                     100    100   
Noho                                   100                     100    100   
Roosevelt Island                        28                      28     28   
Soho                                   100                     100    100   
Stuyvesant Town                         17                      17     17   
Sutton Place                           100                     100    100   
Tribeca                                 97                      97     97   
Tudor City                              79                      79     79   
Turtle Bay                             100                     100    100   
Upper East Side                        100                     100    100   
Upper West Side                        100                     100    100   
Washington Heights                      93                      93     93   
West Village                           100                     100    100   
Yorkville                              100                     100    100   

                     Venue Latitude  Venue Longitude  Venue Category  
Neighborhood                                                          
Battery Park City                87               87              87  
Carnegie Hill                    93               93              93  
Central Harlem                   42               42              42  
Chelsea                         100              100             100  
Chinatown                       100              100             100  
Civic Center                    100              100             100  
Clinton                         100              100             100  
East Harlem                      39               39              39  
East Village                    100              100             100  
Finan

In [39]:
print('There are {} uniques categories.'.format(len(manhattan_venues['Venue Category'].unique())))

There are 330 uniques categories.


In [40]:
print('There are {} neighborhoods in Manhattan.'.format(len(manhattan_venues['Neighborhood'].unique())))

There are 40 neighborhoods in Manhattan.


In [41]:
# one hot encoding
manhattan_onehot = pd.get_dummies(manhattan_venues[['Venue Category']], prefix="", prefix_sep="")

# add neighborhood column back to dataframe
manhattan_onehot['Neighborhood'] = manhattan_venues['Neighborhood'] 

# move neighborhood column to the first column
fixed_columns = [manhattan_onehot.columns[-1]] + list(manhattan_onehot.columns[:-1])
manhattan_onehot = manhattan_onehot[fixed_columns]

manhattan_onehot.head()

Neighborhood  Accessories Store  Acupuncturist  Adult Boutique  \
0  Marble Hill                  0              0               0   
1  Marble Hill                  0              0               0   
2  Marble Hill                  0              0               0   
3  Marble Hill                  0              0               0   
4  Marble Hill                  0              0               0   

   Afghan Restaurant  African Restaurant  American Restaurant  Antique Shop  \
0                  0                   0                    0             0   
1                  0                   0                    0             0   
2                  0                   0                    0             0   
3                  0                   0                    0             0   
4                  0                   0                    0             0   

   Arcade  Arepa Restaurant  Argentinian Restaurant  Art Gallery  Art Museum  \
0       0                 0                       0            0           0   
1       0                 0                       0            0           0   
2       0                 0                       0            0           0   
3       0                 0                       0            0           0   
4       0                 0                       0            0           0   

   Arts & Crafts Store  Asian Restaurant  Athletics & Sports  Auditorium  \
0                    0                 0                   0           0   
1                    0                 0                   0           0   
2                    0                 0                   0           0   
3                    0                 0                   0           0   
4                    0                 0                   0           0   

   Australian Restaurant  Austrian Restaurant  BBQ Joint  Baby Store  \
0                      0                    0          0           0   
1                      0                    0          0           0   
2                      0                    0          0           0   
3                      0                    0          0           0   
4                      0                    0          0           0   

   Bagel Shop  Bakery  Bank  Bar  Baseball Field  Basketball Court  \
0           0       0     0    0               0                 0   
1           0       0     0    0               0                 0   
2           0       0     0    0               0                 0   
3           0       0     0    0               0                 0   
4           0       0     0    0               0                 0   

   Bed & Breakfast  Beer Bar  Beer Garden  Beer Store  \
0                0         0            0           0   
1                0         0            0           0   
2                0         0            0           0   
3                0         0            0           0   
4                0         0            0           0   

   Bike Rental / Bike Share  Bike Shop  Bike Trail  Bistro  Board Shop  \
0                         0          0           0       0           0   
1                         0          0           0       0           0   
2                         0          0           0       0           0   
3                         0          0           0       0           0   
4                         0          0           0       0           0   

   Boat or Ferry  Bookstore  Boutique  Boxing Gym  Brazilian Restaurant  \
0              0          0         0           0                     0   
1              0          0         0           0                     0   
2              0          0         0           0                     0   
3              0          0         0           0                     0   
4              0          0         0           0                     0   

   Breakfast Spot  Bridal Shop  Bridge  Bubble Tea Shop  Building  \
0               0 

In [42]:
manhattan_onehot.shape

(3279, 331)

In [43]:
manhattan_grouped = manhattan_onehot.groupby(['Neighborhood']).mean().reset_index()
manhattan_grouped.head()

Neighborhood  Accessories Store  Acupuncturist  Adult Boutique  \
0  Battery Park City                0.0            0.0             0.0   
1      Carnegie Hill                0.0            0.0             0.0   
2     Central Harlem                0.0            0.0             0.0   
3            Chelsea                0.0            0.0             0.0   
4          Chinatown                0.0            0.0             0.0   

   Afghan Restaurant  African Restaurant  American Restaurant  Antique Shop  \
0                0.0            0.000000             0.011494           0.0   
1                0.0            0.000000             0.010753           0.0   
2                0.0            0.071429             0.047619           0.0   
3                0.0            0.000000             0.030000           0.0   
4                0.0            0.000000             0.040000           0.0   

   Arcade  Arepa Restaurant  Argentinian Restaurant  Art Gallery  Art Museum  \
0     0.0               0.0                     0.0      0.00000    0.000000   
1     0.0               0.0                     0.0      0.00000    0.010753   
2     0.0               0.0                     0.0      0.02381    0.000000   
3     0.0               0.0                     0.0      0.02000    0.000000   
4     0.0               0.0                     0.0      0.00000    0.000000   

   Arts & Crafts Store  Asian Restaurant  Athletics & Sports  Auditorium  \
0                  0.0              0.00            0.011494    0.011494   
1                  0.0              0.00            0.000000    0.000000   
2                  0.0              0.00            0.000000    0.000000   
3                  0.0              0.01            0.000000    0.000000   
4                  0.0              0.02            0.000000    0.000000   

   Australian Restaurant  Austrian Restaurant  BBQ Joint  Baby Store  \
0                    0.0                 0.00   0.022989         0.0   
1                    0.0                 0.00   0.000000         0.0   
2                    0.0                 0.00   0.023810         0.0   
3                    0.0                 0.00   0.000000         0.0   
4                    0.0                 0.01   0.000000         0.0   

   Bagel Shop    Bakery      Bank       Bar  Baseball Field  Basketball Court  \
0    0.000000  0.011494  0.000000  0.000000             0.0               0.0   
1    0.010753  0.032258  0.010753  0.021505             0.0               0.0   
2    0.023810  0.000000  0.000000  0.047619             0.0               0.0   
3    0.010000  0.050000  0.000000  0.010000             0.0               0.0   
4    0.000000  0.030000  0.000000  0.020000             0.0               0.0   

   Bed & Breakfast  Beer Bar  Beer Garden  Beer Store  \
0              0.0   0.00000     0.011494         0.0   
1              0.0   0.00000     0.000000         0.0   
2              0.0   0.02381     0.000000         0.0   
3              0.0   0.01000     0.000000         0.0   
4              0.0   0.00000     0.000000         0.0   

   Bike Rental / Bike Share  Bike Shop  Bike Trail    Bistro  Board Shop  \
0                       0.0       0.00         0.0  0.011494         0.0   
1                       0.0       0.00         0.0  0.000000         0.0   
2                       0.0       0.00         0.0  0.000000         0.0   
3                       0.0       0.00         0.0  0.000000         0.0   
4                       0.0       0.01         0.0  0.000000         0.0   

   Boat or Ferry  Bookstore  Boutique  Boxing Gym  Brazilian Restaurant  \
0       0.034483   0.000000   0.00000         0.0                   0.0   
1       0.000000   0.032258   0.00000         0.0                   0.0   
2       0.000000   0.023810   0.02381         0.0                   0.0   
3       0.000000   0.020000   0.00000         0.0                   0.0   
4       0.000000   0.000000   0.02000         0.0      

In [0]:
venue_categories_corr = manhattan_grouped.corr()[['Frozen Yogurt Shop']]

In [45]:
corr_with_froyoshop = venue_categories_corr[venue_categories_corr['Frozen Yogurt Shop'].map(abs)>0.65]
corr_with_froyoshop

Frozen Yogurt Shop
Bistro                         0.653417
Empanada Restaurant            0.956948
Frozen Yogurt Shop             1.000000
History Museum                 0.671787
Lounge                         0.802493
Restaurant                     0.685575
Veterinarian                   0.956948

In [46]:
df = manhattan_grouped[['Neighborhood'] + list(corr_with_froyoshop.index)]
df.set_index('Neighborhood', inplace=True)
df

Bistro  Empanada Restaurant  Frozen Yogurt Shop  \
Neighborhood                                                             
Battery Park City    0.011494             0.000000            0.000000   
Carnegie Hill        0.000000             0.000000            0.000000   
Central Harlem       0.000000             0.000000            0.000000   
Chelsea              0.000000             0.000000            0.000000   
Chinatown            0.000000             0.000000            0.000000   
Civic Center         0.000000             0.000000            0.000000   
Clinton              0.000000             0.000000            0.000000   
East Harlem          0.000000             0.000000            0.000000   
East Village         0.000000             0.000000            0.000000   
Financial District   0.000000             0.000000            0.000000   
Flatiron             0.000000             0.000000            0.000000   
Gramercy             0.000000             0.000000            0.000000   
Greenwich Village    0.000000             0.000000            0.000000   
Hamilton Heights     0.000000             0.000000            0.000000   
Hudson Yards         0.000000             0.000000            0.000000   
Inwood               0.017857             0.017857            0.035714   
Lenox Hill           0.000000             0.000000            0.000000   
Lincoln Square       0.000000             0.000000            0.000000   
Little Italy         0.000000             0.000000            0.000000   
Lower East Side      0.000000             0.000000            0.000000   
Manhattan Valley     0.000000             0.000000            0.000000   
Manhattanville       0.000000             0.000000            0.000000   
Marble Hill          0.000000             0.000000            0.000000   
Midtown              0.000000             0.000000            0.000000   
Midtown South        0.000000             0.000000            0.000000   
Morningside Heights  0.000000             0.000000            0.000000   
Murray Hill          0.000000             0.000000            0.000000   
Noho                 0.000000             0.000000            0.000000   
Roosevelt Island     0.000000             0.000000            0.000000   
Soho                 0.000000             0.000000            0.000000   
Stuyvesant Town      0.000000             0.000000            0.000000   
Sutton Place         0.000000             0.000000            0.000000   
Tribeca              0.000000             0.000000            0.000000   
Tudor City           0.000000             0.000000            0.000000   
Turtle Bay           0.000000             0.000000            0.000000   
Upper East Side      0.000000             0.000000            0.000000   
Upper West Side      0.000000             0.000000            0.000000   
Washington Heights   0.000000             0.000000            0.010753   
West Village         0.010000             0.000000            0.000000   
Yorkville            0.010000             0.000000            0.000000   

                     History Museum    Lounge  Restaurant  Veterinarian  
Neighborhood                                                             
Battery Park City          0.000000  0.000000    0.000000      0.000000  
Carnegie Hill              0.000000  0.000000    0.010753      0.000000  
Central Harlem             0.000000  0.000000    0.000000      0.000000  
Chelsea                    0.000000  0.000000    0.000000      0.000000  
Chinatown                  0.000000  0.000000    0.000000      0.000000  
Civic Center               0.000000  0.010000    0.000000      0.000000  
Clinton                    0.000000  0.020000    0.010000      0.000000  
East Harlem                0.000000  0.000000    0.025641      0.000000  
East Village               0.000000  0.000000    0.000000      0.000000  
Financial District         0.000000  0.000000    0.010000      0.000000  
Flatiron                  

In [47]:
for column in df.columns:
  df[column] = df['Frozen Yogurt Shop']/df[column]
df

/usr/local/lib/python3.6/dist-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  


Bistro  Empanada Restaurant  Frozen Yogurt Shop  \
Neighborhood                                                           
Battery Park City       0.0                  NaN                 NaN   
Carnegie Hill           NaN                  NaN                 NaN   
Central Harlem          NaN                  NaN                 NaN   
Chelsea                 NaN                  NaN                 NaN   
Chinatown               NaN                  NaN                 NaN   
Civic Center            NaN                  NaN                 NaN   
Clinton                 NaN                  NaN                 NaN   
East Harlem             NaN                  NaN                 NaN   
East Village            NaN                  NaN                 NaN   
Financial District      NaN                  NaN                 NaN   
Flatiron                NaN                  NaN                 NaN   
Gramercy                NaN                  NaN                 NaN   
Greenwich Village       NaN                  NaN                 NaN   
Hamilton Heights        NaN                  NaN                 NaN   
Hudson Yards            NaN                  NaN                 NaN   
Inwood                  2.0                  2.0                 1.0   
Lenox Hill              NaN                  NaN                 NaN   
Lincoln Square          NaN                  NaN                 NaN   
Little Italy            NaN                  NaN                 NaN   
Lower East Side         NaN                  NaN                 NaN   
Manhattan Valley        NaN                  NaN                 NaN   
Manhattanville          NaN                  NaN                 NaN   
Marble Hill             NaN                  NaN                 NaN   
Midtown                 NaN                  NaN                 NaN   
Midtown South           NaN                  NaN                 NaN   
Morningside Heights     NaN                  NaN                 NaN   
Murray Hill             NaN                  NaN                 NaN   
Noho                    NaN                  NaN                 NaN   
Roosevelt Island        NaN                  NaN                 NaN   
Soho                    NaN                  NaN                 NaN   
Stuyvesant Town         NaN                  NaN                 NaN   
Sutton Place            NaN                  NaN                 NaN   
Tribeca                 NaN                  NaN                 NaN   
Tudor City              NaN                  NaN                 NaN   
Turtle Bay              NaN                  NaN                 NaN   
Upper East Side         NaN                  NaN                 NaN   
Upper West Side         NaN                  NaN                 NaN   
Washington Heights      inf                  inf                 1.0   
West Village            0.0                  NaN                 NaN   
Yorkville               0.0                  NaN                 NaN   

                     History Museum     Lounge  Restaurant  Veterinarian  
Neighborhood                                                              
Battery Park City               NaN        NaN         NaN           NaN  
Carnegie Hill                   NaN        NaN         NaN           NaN  
Central Harlem                  NaN        NaN         NaN           NaN  
Chelsea                         NaN        NaN         NaN           NaN  
Chinatown                       NaN        NaN         NaN           NaN  
Civic Center                    NaN        NaN         NaN           NaN  
Clinton                         NaN        NaN         NaN           NaN  
East Harlem                     NaN        NaN         NaN           NaN  
East Village                    NaN        NaN         NaN           NaN  
Financial District              NaN        NaN         NaN           NaN  
Flatiron                        NaN        NaN         NaN           NaN  
Gramercy             

## 3) Paris

In [48]:
filedlurl = r'https://docs.google.com/spreadsheets/d/e/2PACX-1vQXNcKhCQi7jy0AoRKKAafKRqZsLJhwC8jfgBrdHLOqFiPo09A8IHb6C-GpNWI21g/pub?output=xlsx'
paris_data = pd.read_excel(filedlurl, sheet_name='paris by arrondissement')
paris_data

Postal Code Arrondissement  Latitude  Longitude
0         75001       Paris 01   48.8592     2.3417
1         75002       Paris 02   48.8655     2.3426
2         75003       Paris 03   48.8637     2.3615
3         75004       Paris 04   48.8601     2.3507
4         75005       Paris 05   48.8448     2.3471
5         75006       Paris 06   48.8493     2.3300
6         75007       Paris 07   48.8565     2.3210
7         75008       Paris 08   48.8763     2.3183
8         75009       Paris 09   48.8718     2.3399
9         75010       Paris 10   48.8709     2.3561
10        75011       Paris 11   48.8574     2.3795
11        75012       Paris 12   48.8412     2.3876
12        75013       Paris 13   48.8322     2.3561
13        75014       Paris 14   48.8331     2.3264
14        75015       Paris 15   48.8412     2.3003
15        75016       Paris 16   48.8637     2.2769
16        75017       Paris 17   48.8835     2.3219
17        75018       Paris 18   48.8925     2.3444
18        75019       Paris 19   48.8817     2.3822
19        75020       Paris 20   48.8646     2.3984

In [49]:
paris_venues = getNearbyVenues(names=paris_data['Arrondissement'],
                                   latitudes=paris_data['Latitude'],
                                   longitudes=paris_data['Longitude'],
                                   radius=800
                                  )

Paris 01
Paris 02
Paris 03
Paris 04
Paris 05
Paris 06
Paris 07
Paris 08
Paris 09
Paris 10
Paris 11
Paris 12
Paris 13
Paris 14
Paris 15
Paris 16
Paris 17
Paris 18
Paris 19
Paris 20


In [50]:
print(paris_venues.shape)
paris_venues.head()

(1985, 7)


Neighborhood  Neighborhood Latitude  Neighborhood Longitude  \
0     Paris 01                48.8592                  2.3417   
1     Paris 01                48.8592                  2.3417   
2     Paris 01                48.8592                  2.3417   
3     Paris 01                48.8592                  2.3417   
4     Paris 01                48.8592                  2.3417   

                                               Venue  Venue Latitude  \
0                                    Place du Louvre       48.859841   
1  Église Saint-Germain-l'Auxerrois (Église Saint...       48.859520   
2                                          Le Fumoir       48.860424   
3                              Cour Carrée du Louvre       48.860360   
4                                     Place Dauphine       48.856640   

   Venue Longitude    Venue Category  
0         2.340822             Plaza  
1         2.341306            Church  
2         2.340868      Cocktail Bar  
3         2.338543  Pedestrian Plaza  
4         2.342362             Plaza

In [51]:
paris_venues.groupby('Neighborhood').count()

Neighborhood Latitude  Neighborhood Longitude  Venue  \
Neighborhood                                                         
Paris 01                        100                     100    100   
Paris 02                        100                     100    100   
Paris 03                        100                     100    100   
Paris 04                        100                     100    100   
Paris 05                        100                     100    100   
Paris 06                        100                     100    100   
Paris 07                         99                      99     99   
Paris 08                        100                     100    100   
Paris 09                        100                     100    100   
Paris 10                        100                     100    100   
Paris 11                        100                     100    100   
Paris 12                        100                     100    100   
Paris 13                        100                     100    100   
Paris 14                        100                     100    100   
Paris 15                        100                     100    100   
Paris 16                        100                     100    100   
Paris 17                        100                     100    100   
Paris 18                        100                     100    100   
Paris 19                         87                      87     87   
Paris 20                         99                      99     99   

              Venue Latitude  Venue Longitude  Venue Category  
Neighborhood                                                   
Paris 01                 100              100             100  
Paris 02                 100              100             100  
Paris 03                 100              100             100  
Paris 04                 100              100             100  
Paris 05                 100              100             100  
Paris 06                 100              100             100  
Paris 07                  99               99              99  
Paris 08                 100              100             100  
Paris 09                 100              100             100  
Paris 10                 100              100             100  
Paris 11                 100              100             100  
Paris 12                 100              100             100  
Paris 13                 100              100             100  
Paris 14                 100              100             100  
Paris 15                 100              100             100  
Paris 16                 100              100             100  
Paris 17                 100              100             100  
Paris 18                 100              100             100  
Paris 19                  87               87              87  
Paris 20                  99               99              99

In [52]:
print('There are {} uniques categories in Paris.'.format(len(paris_venues['Venue Category'].unique())))

There are 210 uniques categories in Paris.


In [53]:
# one hot encoding
paris_onehot = pd.get_dummies(paris_venues[['Venue Category']], prefix="", prefix_sep="")

# add neighborhood column back to dataframe
paris_onehot['Neighborhood'] = paris_venues['Neighborhood'] 

# move neighborhood column to the first column
fixed_columns = [paris_onehot.columns[-1]] + list(paris_onehot.columns[:-1])
paris_onehot = paris_onehot[fixed_columns]

paris_onehot.head()

Neighborhood  Afghan Restaurant  African Restaurant  Alsatian Restaurant  \
0     Paris 01                  0                   0                    0   
1     Paris 01                  0                   0                    0   
2     Paris 01                  0                   0                    0   
3     Paris 01                  0                   0                    0   
4     Paris 01                  0                   0                    0   

   American Restaurant  Antique Shop  Arcade  Arepa Restaurant  \
0                    0             0       0                 0   
1                    0             0       0                 0   
2                    0             0       0                 0   
3                    0             0       0                 0   
4                    0             0       0                 0   

   Argentinian Restaurant  Art Gallery  Art Museum  Arts & Crafts Store  \
0                       0            0           0                    0   
1                       0            0           0                    0   
2                       0            0           0                    0   
3                       0            0           0                    0   
4                       0            0           0                    0   

   Asian Restaurant  BBQ Joint  Bagel Shop  Bakery  Bar  Basque Restaurant  \
0                 0          0           0       0    0                  0   
1                 0          0           0       0    0                  0   
2                 0          0           0       0    0                  0   
3                 0          0           0       0    0                  0   
4                 0          0           0       0    0                  0   

   Beer Bar  Beer Garden  Beer Store  Bistro  Bookstore  Boutique  Brasserie  \
0         0            0           0       0          0         0          0   
1         0            0           0       0          0         0          0   
2         0            0           0       0          0         0          0   
3         0            0           0       0          0         0          0   
4         0            0           0       0          0         0          0   

   Brazilian Restaurant  Breakfast Spot  Breton Restaurant  Brewery  Bridge  \
0                     0               0                  0        0       0   
1                     0               0                  0        0       0   
2                     0               0                  0        0       0   
3                     0               0                  0        0       0   
4                     0               0                  0        0       0   

   Bubble Tea Shop  Burger Joint  Café  Cajun / Creole Restaurant  \
0                0             0     0                          0   
1                0             0     0                          0   
2                0             0     0                          0   
3                0             0     0                          0   
4                0             0     0                          0   

   Cambodian Restaurant  Canal  Candy Store  Caribbean Restaurant  Cemetery  \
0                     0      0            0                     0         0   
1                     0      0            0                     0         0   
2                     0      0            0                     0         0   
3                     0      0            0                     0         0   
4                     0      0            0                     0         0   

   Cheese Shop  Chinese Restaurant  Chocolate Shop  Church  Clothing Store  \
0            0                   0               0       0               0   
1            0                   0               0       1               0   
2            0                   0               0       0               0   
3            0                   0               0       0

In [54]:
paris_onehot.shape

(1985, 211)

## 4) Cities

In [0]:
manhattan_onehot.insert(0, 'City', 'Manhattan', True)
paris_onehot.insert(0, 'City', 'Paris', True)
old_toronto_onehot.insert(0, 'City', 'Old Toronto', True)

In [0]:
columns_union = list(map(list, [manhattan_onehot.columns, paris_onehot.columns, old_toronto_onehot.columns]))

In [57]:
venues_union = pd.concat([manhattan_onehot, paris_onehot, old_toronto_onehot], 
                         axis = 0, ignore_index=True)
venues_union.fillna(0, inplace=True)
venues_union.head()

City Neighborhood  Accessories Store  Acupuncturist  Adult Boutique  \
0  Manhattan  Marble Hill                0.0            0.0             0.0   
1  Manhattan  Marble Hill                0.0            0.0             0.0   
2  Manhattan  Marble Hill                0.0            0.0             0.0   
3  Manhattan  Marble Hill                0.0            0.0             0.0   
4  Manhattan  Marble Hill                0.0            0.0             0.0   

   Afghan Restaurant  African Restaurant  American Restaurant  Antique Shop  \
0                  0                 0.0                    0             0   
1                  0                 0.0                    0             0   
2                  0                 0.0                    0             0   
3                  0                 0.0                    0             0   
4                  0                 0.0                    0             0   

   Arcade  Arepa Restaurant  Argentinian Restaurant  Art Gallery  Art Museum  \
0     0.0                 0                     0.0            0         0.0   
1     0.0                 0                     0.0            0         0.0   
2     0.0                 0                     0.0            0         0.0   
3     0.0                 0                     0.0            0         0.0   
4     0.0                 0                     0.0            0         0.0   

   Arts & Crafts Store  Asian Restaurant  Athletics & Sports  Auditorium  \
0                    0                 0                 0.0         0.0   
1                    0                 0                 0.0         0.0   
2                    0                 0                 0.0         0.0   
3                    0                 0                 0.0         0.0   
4                    0                 0                 0.0         0.0   

   Australian Restaurant  Austrian Restaurant  BBQ Joint  Baby Store  \
0                    0.0                  0.0          0         0.0   
1                    0.0                  0.0          0         0.0   
2                    0.0                  0.0          0         0.0   
3                    0.0                  0.0          0         0.0   
4                    0.0                  0.0          0         0.0   

   Bagel Shop  Bakery  Bank  Bar  Baseball Field  Basketball Court  \
0           0       0   0.0    0             0.0               0.0   
1           0       0   0.0    0             0.0               0.0   
2           0       0   0.0    0             0.0               0.0   
3           0       0   0.0    0             0.0               0.0   
4           0       0   0.0    0             0.0               0.0   

   Bed & Breakfast  Beer Bar  Beer Garden  Beer Store  \
0              0.0         0          0.0           0   
1              0.0         0          0.0           0   
2              0.0         0          0.0           0   
3              0.0         0          0.0           0   
4              0.0         0          0.0           0   

   Bike Rental / Bike Share  Bike Shop  Bike Trail  Bistro  Board Shop  \
0                       0.0        0.0         0.0       0         0.0   
1                       0.0        0.0         0.0       0         0.0   
2                       0.0        0.0         0.0       0         0.0   
3                       0.0        0.0         0.0       0         0.0   
4                       0.0        0.0         0.0       0         0.0   

   Boat or Ferry  Bookstore  Boutique  Boxing Gym  Brazilian Restaurant  \
0            0.0          0         0         0.0                     0   
1            0.0          0         0         0.0                     0   
2            0.0          0         0         0.0                     0   
3            0.0          0         0         0.0                     0   
4            0.0          0         0         0.0                     0   

   Breakfast Spot  Bridal S

In [58]:
venues_union.shape

(7386, 407)

In [59]:
#venues grouped by neighborhood
venues_grouped = venues_union.groupby(['City','Neighborhood']).mean().reset_index()
venues_grouped.head()

City       Neighborhood  Accessories Store  Acupuncturist  \
0  Manhattan  Battery Park City                0.0            0.0   
1  Manhattan      Carnegie Hill                0.0            0.0   
2  Manhattan     Central Harlem                0.0            0.0   
3  Manhattan            Chelsea                0.0            0.0   
4  Manhattan          Chinatown                0.0            0.0   

   Adult Boutique  Afghan Restaurant  African Restaurant  American Restaurant  \
0             0.0                0.0            0.000000             0.011494   
1             0.0                0.0            0.000000             0.010753   
2             0.0                0.0            0.071429             0.047619   
3             0.0                0.0            0.000000             0.030000   
4             0.0                0.0            0.000000             0.040000   

   Antique Shop  Arcade  Arepa Restaurant  Argentinian Restaurant  \
0           0.0     0.0               0.0                     0.0   
1           0.0     0.0               0.0                     0.0   
2           0.0     0.0               0.0                     0.0   
3           0.0     0.0               0.0                     0.0   
4           0.0     0.0               0.0                     0.0   

   Art Gallery  Art Museum  Arts & Crafts Store  Asian Restaurant  \
0      0.00000    0.000000                  0.0              0.00   
1      0.00000    0.010753                  0.0              0.00   
2      0.02381    0.000000                  0.0              0.00   
3      0.02000    0.000000                  0.0              0.01   
4      0.00000    0.000000                  0.0              0.02   

   Athletics & Sports  Auditorium  Australian Restaurant  Austrian Restaurant  \
0            0.011494    0.011494                    0.0                 0.00   
1            0.000000    0.000000                    0.0                 0.00   
2            0.000000    0.000000                    0.0                 0.00   
3            0.000000    0.000000                    0.0                 0.00   
4            0.000000    0.000000                    0.0                 0.01   

   BBQ Joint  Baby Store  Bagel Shop    Bakery      Bank       Bar  \
0   0.022989         0.0    0.000000  0.011494  0.000000  0.000000   
1   0.000000         0.0    0.010753  0.032258  0.010753  0.021505   
2   0.023810         0.0    0.023810  0.000000  0.000000  0.047619   
3   0.000000         0.0    0.010000  0.050000  0.000000  0.010000   
4   0.000000         0.0    0.000000  0.030000  0.000000  0.020000   

   Baseball Field  Basketball Court  Bed & Breakfast  Beer Bar  Beer Garden  \
0             0.0               0.0              0.0   0.00000     0.011494   
1             0.0               0.0              0.0   0.00000     0.000000   
2             0.0               0.0              0.0   0.02381     0.000000   
3             0.0               0.0              0.0   0.01000     0.000000   
4             0.0               0.0              0.0   0.00000     0.000000   

   Beer Store  Bike Rental / Bike Share  Bike Shop  Bike Trail    Bistro  \
0         0.0                       0.0       0.00         0.0  0.011494   
1         0.0                       0.0       0.00         0.0  0.000000   
2         0.0                       0.0       0.00         0.0  0.000000   
3         0.0                       0.0       0.00         0.0  0.000000   
4         0.0                       0.0       0.01         0.0  0.000000   

   Board Shop  Boat or Ferry  Bookstore  Boutique  Boxing Gym  \
0         0.0       0.034483   0.000000   0.00000         0.0   
1         0.0       0.000000   0.032258   0.00000         0.0   
2         0.0       0.000000   0.023810   0.02381         0.0   
3         0.0       0.000000   0.020000   0.00000         0.0   
4         0.0       0.000000   0.000000   0.02000         0.0   

   Brazilian Restaurant  Breakfast Spot  Brid

In [75]:
freq_by_city = venues_union.groupby('City').mean()
freq_by_city

Accessories Store  Acupuncturist  Adult Boutique  \
City                                                            
Manhattan              0.00122       0.000305        0.000305   
Old Toronto            0.00000       0.000000        0.000000   
Paris                  0.00000       0.000000        0.000000   

             Afghan Restaurant  African Restaurant  American Restaurant  \
City                                                                      
Manhattan             0.000305            0.000915             0.027142   
Old Toronto           0.000471            0.000000             0.012253   
Paris                 0.000504            0.002519             0.003526   

             Antique Shop    Arcade  Arepa Restaurant  Argentinian Restaurant  \
City                                                                            
Manhattan        0.000305  0.000305          0.000305                0.001220   
Old Toronto      0.001885  0.000000          0.000471                0.000000   
Paris            0.000504  0.000504          0.000504                0.002015   

             Art Gallery  Art Museum  Arts & Crafts Store  Asian Restaurant  \
City                                                                          
Manhattan       0.007319    0.001220             0.001220          0.004270   
Old Toronto     0.006598    0.000000             0.004713          0.005655   
Paris           0.009572    0.006045             0.000504          0.006045   

             Athletics & Sports  Auditorium  Australian Restaurant  \
City                                                                 
Manhattan              0.000915    0.000305                0.00122   
Old Toronto            0.000943    0.000000                0.00000   
Paris                  0.000000    0.000000                0.00000   

             Austrian Restaurant  BBQ Joint  Baby Store  Bagel Shop    Bakery  \
City                                                                            
Manhattan                0.00061   0.001220    0.000305    0.006709  0.022568   
Old Toronto              0.00000   0.003299    0.000471    0.002356  0.019321   
Paris                    0.00000   0.001008    0.000000    0.001511  0.031738   

                 Bank       Bar  Baseball Field  Basketball Court  \
City                                                                
Manhattan    0.002440  0.017993        0.000305          0.000305   
Old Toronto  0.005184  0.022620        0.000000          0.000000   
Paris        0.000000  0.038287        0.000000          0.000000   

             Bed & Breakfast  Beer Bar  Beer Garden  Beer Store  \
City                                                              
Manhattan           0.000305  0.002745     0.000915    0.000915   
Old Toronto         0.000000  0.010368     0.000000    0.002356   
Paris               0.000000  0.005038     0.001511    0.002519   

             Bike Rental / Bike Share  Bike Shop  Bike Trail    Bistro  \
City                                                                     
Manhattan                    0.000915   0.000305    0.000305  0.001220   
Old Toronto                  0.000000   0.000000    0.000000  0.002356   
Paris                        0.000000   0.000000    0.000000  0.025189   

             Board Shop  Boat or Ferry  Bookstore  Boutique  Boxing Gym  \
City                                                                      
Manhattan       0.00061       0.002135   0.007929  0.004575     0.00183   
Old Toronto     0.00000       0.000943   0.010368  0.001885     0.00000   
Paris           0.00000       0.000000   0.011587  0.001511     0.00000   

             Brazilian Restaurant  Breakfast Spot  Bridal Shop    Bridge  \
City                                                                       
Manhattan                0.000610        0.002440      0.00061  0.000610   
Old Toronto              0.001414        0.009896      0.00000  0.000000   
Paris              

In [0]:
from scipy.stats import ks_2samp

freq_manhattan = freq_by_city.values[0]
freq_old_toronto = freq_by_city.values[1]
freq_paris = freq_by_city.values[2]

In [77]:
ks_2samp(freq_manhattan, freq_old_toronto)

Ks_2sampResult(statistic=0.20246913580246914, pvalue=1.1211445674119701e-07)

In [78]:
ks_2samp(freq_manhattan, freq_paris)

Ks_2sampResult(statistic=0.2962962962962963, pvalue=4.416875422592007e-16)

In [0]:
def return_most_common_venues(row, num_top_venues):
    row_categories = row.iloc[2:]
    row_categories_sorted = row_categories.sort_values(ascending=False)
    return row_categories_sorted.index.values[0:num_top_venues]

In [65]:
num_top_venues = 10

indicators = ['st', 'nd', 'rd']

# create columns according to number of top venues
columns = ['Neighborhood']
for ind in np.arange(num_top_venues):
    try:
        columns.append('{}{} Most Common Venue'.format(ind+1, indicators[ind]))
    except:
        columns.append('{}th Most Common Venue'.format(ind+1))

# create a new dataframe
neighborhoods_venues_sorted = pd.DataFrame(columns=columns)
neighborhoods_venues_sorted['Neighborhood'] = venues_grouped['Neighborhood']

for ind in np.arange(venues_grouped.shape[0]):
    neighborhoods_venues_sorted.iloc[ind, 1:] = return_most_common_venues(venues_grouped.iloc[ind, :], num_top_venues)

neighborhoods_venues_sorted.head()

Neighborhood 1st Most Common Venue 2nd Most Common Venue  \
0  Battery Park City                  Park           Coffee Shop   
1      Carnegie Hill           Coffee Shop                  Café   
2     Central Harlem    African Restaurant     French Restaurant   
3            Chelsea           Coffee Shop                Bakery   
4          Chinatown    Chinese Restaurant          Cocktail Bar   

  3rd Most Common Venue 4th Most Common Venue 5th Most Common Venue  \
0                 Hotel             Wine Shop         Memorial Site   
1           Pizza Place                   Gym  Gym / Fitness Center   
2    Chinese Restaurant   American Restaurant                   Bar   
3    Italian Restaurant        Ice Cream Shop                 Hotel   
4   American Restaurant                   Spa          Dessert Shop   

  6th Most Common Venue 7th Most Common Venue  8th Most Common Venue  \
0                   Gym        Clothing Store          Boat or Ferry   
1             Bookstore             Wine Shop    Japanese Restaurant   
2    Seafood Restaurant          Cycle Studio    Fried Chicken Joint   
3   American Restaurant   Japanese Restaurant              Wine Shop   
4     Hotpot Restaurant          Optical Shop  Vietnamese Restaurant   

  9th Most Common Venue 10th Most Common Venue  
0         Shopping Mall            Pizza Place  
1     French Restaurant                 Bakery  
2              Beer Bar           Dessert Shop  
3             Bookstore           Cocktail Bar  
4                Bakery     Salon / Barbershop

In [66]:
# set number of clusters
kclusters = 3

venues_grouped_clustering = venues_grouped.drop(['City','Neighborhood'], 1)

# run k-means clustering
kmeans = KMeans(n_clusters=kclusters, random_state=1).fit(venues_grouped_clustering)

# check cluster labels generated for each row in the dataframe
kmeans.labels_

array([1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
       1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
       1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 2, 1, 2, 1, 1, 1, 1, 1,
       2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0, 1, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], dtype=int32)

In [0]:
# add clustering labels
venues_grouped.insert(0, 'Cluster Labels', kmeans.labels_)

In [68]:
venues_grouped

Cluster Labels         City  \
0                1    Manhattan   
1                1    Manhattan   
2                1    Manhattan   
3                1    Manhattan   
4                1    Manhattan   
5                1    Manhattan   
6                1    Manhattan   
7                1    Manhattan   
8                1    Manhattan   
9                1    Manhattan   
10               1    Manhattan   
11               1    Manhattan   
12               1    Manhattan   
13               1    Manhattan   
14               1    Manhattan   
15               1    Manhattan   
16               1    Manhattan   
17               1    Manhattan   
18               1    Manhattan   
19               1    Manhattan   
20               1    Manhattan   
21               1    Manhattan   
22               1    Manhattan   
23               1    Manhattan   
24               1    Manhattan   
25               1    Manhattan   
26               1    Manhattan   
27               1    Manhattan   
28               1    Manhattan   
29               1    Manhattan   
30               1    Manhattan   
31               1    Manhattan   
32               1    Manhattan   
33               1    Manhattan   
34               1    Manhattan   
35               1    Manhattan   
36               1    Manhattan   
37               1    Manhattan   
38               1    Manhattan   
39               1    Manhattan   
40               1  Old Toronto   
41               1  Old Toronto   
42               1  Old Toronto   
43               1  Old Toronto   
44               1  Old Toronto   
45               1  Old Toronto   
46               1  Old Toronto   
47               1  Old Toronto   
48               1  Old Toronto   
49               1  Old Toronto   
50               1  Old Toronto   
51               1  Old Toronto   
52               2  Old Toronto   
53               1  Old Toronto   
54               1  Old Toronto   
55               1  Old Toronto   
56               1  Old Toronto   
57               1  Old Toronto   
58               2  Old Toronto   
59               1  Old Toronto   
60               2  Old Toronto   
61               1  Old Toronto   
62               1  Old Toronto   
63               1  Old Toronto   
64               1  Old Toronto   
65               1  Old Toronto   
66               2  Old Toronto   
67               1  Old Toronto   
68               1  Old Toronto   
69               1  Old Toronto   
70               1  Old Toronto   
71               1  Old Toronto   
72               1  Old Toronto   
73               1  Old Toronto   
74               1  Old Toronto   
75               1  Old Toronto   
76               1  Old Toronto   
77               1  Old Toronto   
78               1  Old Toronto   
79               0        Paris   
80               0        Paris   
81               1        Paris   
82               0        Paris   
83               0        Paris   
84               0        Paris   
85               0        Paris   
86               0        Paris   
87               0        Paris   
88               0        Paris   
89               0        Paris   
90               0        Paris   
91               0        Paris   
92               0        Paris   
93               0        Paris   
94               0        Paris   
95               0        Paris   
96               0        Paris   
97               0        Paris   
98               0        Paris   

                                         Neighborhood  Accessories Store  \
0                                   Battery Park City           0.000000   
1                                       Carnegie Hill           0.000000   
2                                      Central Harlem           0.000000   
3                                             Chelsea           0.000000   
4                                           Chinatown           0.000000   
5                                        Civic 

In [69]:
# merge clustered neighborhood with most common venues for each neighborhood
cities_neighborhoods_clustered = venues_grouped[['Cluster Labels', 'City', 'Neighborhood']].join(neighborhoods_venues_sorted.set_index('Neighborhood'), on='Neighborhood').sort_values(by=['Cluster Labels', 'City'])
cities_neighborhoods_clustered # check the last columns!

Cluster Labels         City  \
79               0        Paris   
80               0        Paris   
82               0        Paris   
83               0        Paris   
84               0        Paris   
85               0        Paris   
86               0        Paris   
87               0        Paris   
88               0        Paris   
89               0        Paris   
90               0        Paris   
91               0        Paris   
92               0        Paris   
93               0        Paris   
94               0        Paris   
95               0        Paris   
96               0        Paris   
97               0        Paris   
98               0        Paris   
0                1    Manhattan   
1                1    Manhattan   
2                1    Manhattan   
3                1    Manhattan   
4                1    Manhattan   
5                1    Manhattan   
6                1    Manhattan   
7                1    Manhattan   
8                1    Manhattan   
9                1    Manhattan   
10               1    Manhattan   
11               1    Manhattan   
12               1    Manhattan   
13               1    Manhattan   
14               1    Manhattan   
15               1    Manhattan   
16               1    Manhattan   
17               1    Manhattan   
18               1    Manhattan   
19               1    Manhattan   
20               1    Manhattan   
21               1    Manhattan   
22               1    Manhattan   
23               1    Manhattan   
24               1    Manhattan   
25               1    Manhattan   
26               1    Manhattan   
27               1    Manhattan   
28               1    Manhattan   
29               1    Manhattan   
30               1    Manhattan   
31               1    Manhattan   
32               1    Manhattan   
33               1    Manhattan   
34               1    Manhattan   
35               1    Manhattan   
36               1    Manhattan   
37               1    Manhattan   
38               1    Manhattan   
39               1    Manhattan   
40               1  Old Toronto   
41               1  Old Toronto   
42               1  Old Toronto   
43               1  Old Toronto   
44               1  Old Toronto   
45               1  Old Toronto   
46               1  Old Toronto   
47               1  Old Toronto   
48               1  Old Toronto   
49               1  Old Toronto   
50               1  Old Toronto   
51               1  Old Toronto   
53               1  Old Toronto   
54               1  Old Toronto   
55               1  Old Toronto   
56               1  Old Toronto   
57               1  Old Toronto   
59               1  Old Toronto   
61               1  Old Toronto   
62               1  Old Toronto   
63               1  Old Toronto   
64               1  Old Toronto   
65               1  Old Toronto   
67               1  Old Toronto   
68               1  Old Toronto   
69               1  Old Toronto   
70               1  Old Toronto   
71               1  Old Toronto   
72               1  Old Toronto   
73               1  Old Toronto   
74               1  Old Toronto   
75               1  Old Toronto   
76               1  Old Toronto   
77               1  Old Toronto   
78               1  Old Toronto   
81               1        Paris   
52               2  Old Toronto   
58               2  Old Toronto   
60               2  Old Toronto   
66               2  Old Toronto   

                                         Neighborhood 1st Most Common Venue  \
79                                           Paris 01     French Restaurant   
80                                           Paris 02     French Restaurant   
82                                           Paris 04     French Restaurant   
83                                           Paris 05     French Restaurant   
84                                           Paris 06     French Restaurant   
85                           

In [70]:
cities_neighborhoods_clustered.groupby(['Cluster Labels'])['City'].value_counts()

Cluster Labels  City       
0               Paris          19
1               Manhattan      40
                Old Toronto    35
                Paris           1
2               Old Toronto     4
Name: City, dtype: int64

# 3. Report

**(3) Methodology**

Import the ***Kolmogorov-Smirnov test***  to compare 1-D arrays of the venue category frequencies in the three cities (venues grouped by Toronto, Manhattan, Paris).
The Kolmogorov-Smirnov test (KS-test) tries to determine if two datasets differ significantly. The KS-test has the advantage of making no assumption about the distribution of data. (Technically speaking it is non-parametric and distribution free.)

Aggregate the venues by neighborhood in the 3 cities, ***K-means cluster*** the neighborhoods into 3 clusters. See how they are clustered and counts in terms of the cities.

 <br/>

**(4) Results**

The KS-test result between venues frequencies in Manhattan and that in Old Toronto is 0.20246913580246914, with p-value of 1.1211445674119701e-07 indicating the two places are ***significantly different*** in terms of venues located in their neighborhoods.

Similarly, The KS-test result between venues frequencies in Manhattan and that in Paris is 0.2962962962962963, suggesting ***Paris is significantly different from Manhattan*** in terms of venues as well.

In K-means clustering, however, 19 out of the 20 (95%) Paris neighborhoods are in Cluster 0, while 35 out of the 39 (90%) old Toronto neighborhoods are fall together with all the 40 (100%) Manhattan neighborhoods in Cluster 1. The remaining 4 Toronto neighborhoods are in Cluster 2.

The clustering shows Paris can be ***clearly separated***, while all neighborhoods of Manhattan are also ***clustered together*** but with the majority of old Toronto neighborhoods.
 
 <br/>
 
**(5) Discussion**

As the venue categories obtained from Foursquare are not exactly the same, 0 is filled in to complete the union set. However, as the venue categorization may be different in the database through the countries, some errors or loss of similarity may be caused here, i.e. similar venues may be classified under a different “venue category” name.
 
<br/>

**(6) Conclusion**

Based on the data of venues in the neighborhoods, ***Manhattan seems dissimilar to both old Toronto and Paris*** according to the non-parametric test - Kolmogorov-Smirnov test results. ***But compared to Paris, Manhattan may be more like old Toronto.***
 
